### DISCUSSION TITLE
How can these two rules be executed without circular error

### DISCUSSION THREAD

**Discussion by RouGang** [hyperjumptech/grule-rule-engine#274] [Q&A]
**Is your feature request related to a problem? Please describe.**

rule CheckValues "Check the default values" salience 8 {
	when
		MF.IntAttribute == 123 && MF.StringAttribute == "Some string value"
	then
		MF.WhatToSay = MF.GetWhatToSay("Hello Grule!");
		Retract("CheckValues");
}

rule CheckValuesV2 "Check the alternative" salience 9 {
	when
		MF.IntAttribute == 123 && MF.StringAttribute == "Some string value"
	then
		MF.WhatToSay = MF.GetWhatToSay("Hello Grule v2!");
		Retract("CheckValuesV2");
}

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by caicaispace** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by newm4n** [comment]
Hi @RouGang 

I don't really get what you're asking ?
Could you elaborate more ? Like.. Tell us what you're trying to do ?

**Comment by newm4n** [comment]
@RouGang  lets discuss it here.
